### PR TITLE
Tech/473/update plop generator

### LINKED ISSUE #473: Adapt plop file generator to new structure
# Feature request

## Description

As the folder structure changed with #178, the [plopfile.js](https://github.com/MaibornWolff/codecharta/blob/master/visualization/plopfile.js) file generator needs to get updated. It should provide three different options to generate source-files:
- `state` service (needs to be adapted from the templates for the old core-modules). But we don't want the angular-modules anymore. Now we only use multiple services with each a test-file within one `state` modele e.x.
- `ui` component (does not need to be updated)
- `util` class (create new template)

### PR DESCRIPTION
**Pull request by alschmut**
# Tech/473/update plop generator

closes #473 
Merge permission: CC-Member

## Description

- Update `visualization/plopfile.ts` and `visualization/plop-templates` to adapt to new project structure with 
    - `state` services
    - `ui` modules
    - `util` static classes
- Update test templates to our new test-file-structure

## Question
Should we update the `CHANGELOG.md` for this PR as its actually not for the end user?

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [ ] **Update the CHANGELOG.md** with any changes/additions made
* [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/plop-templates/service.spec.ts.hbs [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-    });
+        restartSystem()
+        rebuildService()
+    })
 
-    it("should do something", ()=>{
-        
-    });
+    function restartSystem() {
+        // initialise injected services and used variables
```
Comment (NearW) [inline visualization/plop-templates/service.spec.ts.hbs]: Instantiate Module as well?

File: visualization/plop-templates/service.spec.ts.hbs [hunk 2]
```diff
@@ -1,15 +1,25 @@
-import { {{properCase name}}Service } from "./{{camelCase name}}.service";
+import { {{properCase name}}Service } from "./{{camelCase name}}.service"
 
 describe("{{properCase name}}Service", () => {
 
-    let {{camelCase name}}Service: {{properCase name}}Service;
+    let {{camelCase name}}Service: {{properCase name}}Service
 
     beforeEach(function() {
```
Comment (NearW) [inline visualization/plop-templates/service.spec.ts.hbs]: use arrow-function `() => {}` instead of `function()`

Comment (alschmut) [reply visualization/plop-templates/service.spec.ts.hbs]: guter Einwand, hab ich übersehen

File: visualization/plop-templates/service.spec.ts.hbs (hunk 1 above)
Comment (alschmut) [reply visualization/plop-templates/service.spec.ts.hbs]: I added it

**Pull request merged**